app/Models/Payment.php

```
@@ -135,84 +135,85 @@ public function getCollectionPaymentByBranch($date)
+                'status' => 'open'
+            ])->lockForUpdate()->first();
+            
+            if ($existingPayment) {
+                \Log::info('Duplicate payment prevented. Existing payment ID: ' . $existingPayment->payment_id);
+                return $existingPayment;
             }
 
-            if ($payment->penalty > 0 && !$payment->penalty_approval_no) {
-                $penalty = $payment->penalty;
-            }
+            $account = LoanAccount::find($request->input('loan_account_id'));
```

PHPMD (2.15.0)

Avoid using static access to class 'App\Models\LoanAccount' in method 'addPayment'. (Clean Code Rules)

(StaticAccess)